### PR TITLE
Support statement timeout

### PR DESCRIPTION
**Pull request by brian-rickman**
The primary motivation for this change is to support a new automated test that will make use of a random sql query generator.  Sometimes the generator produces queries which may run for an excessively long time.  If this occurs, it is preferable to have the offending query canceled after a few minutes so that other queries can be executed.

Added ability to specify statement timeout in database configuration file.  Refactored JdbcConnection by removing unused constructors.  Cleaned up readme and sample configuration file.  Insert the version number into the wrapper script as part of the build so that the correct jar is loaded at runtime.

Tested with a variety of queries and timeout settings to prove that queries which exceed the specified timeout are canceled, and that queries which take less time are allowed to run to completion.

### CODE REVIEW

File: expected-result-generator/bin/generate_results.py [hunk 1]
```diff
@@ -10,8 +10,11 @@
 import argparse
 import subprocess
 import os
+import glob
 
-CLASSPATH = ".:../build/libs/expected-result-generator-all.jar"
+jar_file_list = glob.glob("../build/libs/expected-result-generator-all*.jar")
```
Comment (kokosing) [inline expected-result-generator/bin/generate_results.py]: this probably does not belong to this change... Please mention about that in commit message

File: expected-result-generator/src/main/java/com/teradata/tempto/sql/JdbcConnection.java [hunk 2]
```diff
@@ -75,44 +64,22 @@ public JdbcConnection(String propertiesFileName) throws IOException {
...
-  }
-
-  /**
-   * @param jdbcUrl    jdbc connection URL
-   * @param user       Database user name
-   * @param password   Database password
-   */
-  public JdbcConnection(String jdbcUrl, String user, String password) {
-    initialize(jdbcUrl, user, password);
+    LOGGER.info("timeout {}", timeout);
```
Comment (kokosing) [inline expected-result-generator/src/main/java/com/teradata/tempto/sql/JdbcConnection.java]: join this 4 log statements into one

File: expected-result-generator/src/main/java/com/teradata/tempto/sql/JdbcConnection.java [hunk 3]
```diff
@@ -75,44 +64,22 @@ public JdbcConnection(String propertiesFileName) throws IOException {
...
-    String url = properties.getProperty(JDBC_URL);
-    String user = properties.getProperty(JDBC_USER);
-    String password = properties.getProperty(JDBC_PASSWORD);
-
-    initialize(url, user, password);
-  }
+    jdbcUrl = properties.getProperty(JDBC_URL);
+    user = properties.getProperty(JDBC_USER);
+    password = properties.getProperty(JDBC_PASSWORD);
+    timeout = Integer.parseInt(properties.getProperty(JDBC_TIMEOUT, "600"));
```
Comment (kokosing) [inline expected-result-generator/src/main/java/com/teradata/tempto/sql/JdbcConnection.java]: document that 10 minutes is a default timeout

File: expected-result-generator/src/main/java/com/teradata/tempto/sql/SqlResultGenerator.java [hunk 4]
```diff
@@ -80,10 +67,12 @@ public static void main(String[] args) {
       resultGenerator.generateExpectedResults();
     } catch (ParseException e) {
       usage(options);
+      System.exit(99);
```
Comment (kokosing) [inline expected-result-generator/src/main/java/com/teradata/tempto/sql/SqlResultGenerator.java]: do not use System.exit()... ever

File: expected-result-generator/src/main/java/com/teradata/tempto/sql/SqlResultGenerator.java [hunk 5]
```diff
@@ -80,10 +67,12 @@ public static void main(String[] args) {
...
     } catch (Exception e) {
       LOGGER.error("Caught exception in main", e);
+      System.exit(99);
```
Comment (kokosing) [inline expected-result-generator/src/main/java/com/teradata/tempto/sql/SqlResultGenerator.java]: what does 99 mean?

**Comment by kokosing** [comment]
Please write in the description to what component this change is related to. Also write about what steps you took to make sure it is well tested.

File: expected-result-generator/bin/generate_results.py (hunk 1 above)
Comment (brian-rickman) [inline expected-result-generator/bin/generate_results.py]: Updated description to show that this change is intentional.

Comment (losipiuk) [inline expected-result-generator/bin/generate_results.py]: You should not use * here.  You can easily end up with multiple versions of
jar in the build directory and then you will have messed up classpath.
Generally it would be best to have build target which builds distribution
package with both running script and proper jar file.

File: expected-result-generator/src/main/java/com/teradata/tempto/sql/JdbcConnection.java (hunk 2 above)
Comment (brian-rickman) [inline expected-result-generator/src/main/java/com/teradata/tempto/sql/JdbcConnection.java]: Done

File: expected-result-generator/src/main/java/com/teradata/tempto/sql/JdbcConnection.java (hunk 3 above)
Comment (brian-rickman) [inline expected-result-generator/src/main/java/com/teradata/tempto/sql/JdbcConnection.java]: Done

File: expected-result-generator/src/main/java/com/teradata/tempto/sql/SqlResultGenerator.java (hunk 4 above)
Comment (brian-rickman) [inline expected-result-generator/src/main/java/com/teradata/tempto/sql/SqlResultGenerator.java]: Removed

File: expected-result-generator/src/main/java/com/teradata/tempto/sql/SqlResultGenerator.java (hunk 5 above)
Comment (brian-rickman) [inline expected-result-generator/src/main/java/com/teradata/tempto/sql/SqlResultGenerator.java]: Removed

**Comment by brian-rickman** [comment]
@kokosing Thank you for reviewing my code.  I have addressed all of your comments in my second commit.

File: expected-result-generator/bin/generate_results.py (hunk 1 above)
Comment (brian-rickman) [inline expected-result-generator/bin/generate_results.py]: That does sound like a better technique.  I will look into it.

**Comment by kokosing** [comment]
Thank you. Code looks good to me, modulo @losipiuk comments.
